pypesto/optimize/util.py

```diff
@@ -138,3 +139,38 @@ def bound_n_starts_from_env(n_starts: int):
     )
 
     return n_starts_new
+
+
+def assign_ids(
+    ids: Iterable[str] = None,
+    result: Result = None,
+    n_starts: int = 100,
+) -> Iterable[str]:
+    """
+    Assign ids to starts.
+
+    ids:
+        Ids assigned to the startpoints.
+    result:
+        A result object to append the optimization results to. For example,
+        one might append more runs to a previous optimization. Assign only
+        unique ids.
+    n_starts:
+        Number of starts of the optimizer.
+    """
+    used_ids = set({})
+    if result is not None:
+        used_ids = result.optimize_result.id
+    if ids is None:
+        i = 0
+        ids = [str(j) for j in range(i * n_starts, (i + 1) * n_starts)]
+        while not used_ids.isdisjoint(ids):
+            i += 1
+            ids = [str(j) for j in range(i * n_starts, (i + 1) * n_starts)]
```

Comment:
If `N_starts = len(used_ids)` is the number of previous starts, and `N_starts >> n_starts`, then this might loop many times. Perhaps can use multiples of `N_starts` as the starting point? e.g. `range(i * N_starts, i * N_starts + n_starts)`

Reply:
might be a good idea 👍